utils/gear-replay-cli/src/lib.rs

```diff
@@ -16,103 +16,262 @@
+pub type HashFor<B> = <B as BlockT>::Hash;
+pub type NumberFor<B> = <<B as BlockT>::Header as HeaderT>::Number;
 
 #[derive(Clone, Debug)]
-pub(crate) enum BlockHashOrNumber<B: BlockT> {
+pub enum BlockHashOrNumber<B: BlockT> {
     Hash(HashFor<B>),
     Number(NumberFor<B>),
 }
 
-/// Commands of `gear-replay` CLI
-#[derive(Debug, Subcommand)]
-pub enum Command {
-    ReplayBlock(replay_block::ReplayBlockCmd<Block>),
-    GearRun(gear_run::GearRunCmd<Block>),
+impl<B: BlockT> core::fmt::Display for BlockHashOrNumber<B> {
+    fn fmt(&self, f: &mut core::fmt::Formatter<'_>) -> core::fmt::Result {
+        match self {
+            BlockHashOrNumber::Hash(hash) => {
+                write!(f, "{}", hex::encode(hash.as_ref()))
+            }
+            BlockHashOrNumber::Number(number) => {
+                write!(f, "{}", number)
+            }
+        }
+    }
 }
 
-/// Parameters shared across the subcommands
-#[derive(Clone, Debug, Parser)]
-#[group(skip)]
-pub struct SharedParams {
-    /// The RPC url.
-    #[arg(
-		short,
-		long,
-		value_parser = parse::url,
-		default_value = "wss://archive-rpc.vara.network:443"
-	)]
-    uri: String,
-
-    /// Sets a custom logging filter. Syntax is `<target>=<level>`, e.g. -lsync=debug.
-    ///
-    /// Log levels (least to most verbose) are error, warn, info, debug, and trace.
-    /// By default, all targets log `info`. The global log level can be set with `-l<level>`.
-    #[arg(short = 'l', long, value_name = "NODE_LOG", num_args = 0..)]
-    pub log: Vec<String>,
+impl<Block: BlockT> BlockHashOrNumber<Block> {
+    pub(crate) async fn as_hash(&self, rpc: &WsClient) -> sc_cli::Result<Block::Hash>
+    where
+        Block: DeserializeOwned,
+        Block::Header: DeserializeOwned,
+    {
+        match self {
+            BlockHashOrNumber::Hash(h) => Ok(*h),
+            BlockHashOrNumber::Number(n) => Ok(
+                match ChainApi::<(), Block::Hash, Block::Header, ()>::block_hash(
+                    rpc,
+                    Some(ListOrValue::Value(NumberOrHex::Number(
+                        (*n).try_into()
+                            .map_err(|_| "failed to convert number to block number")?,
+                    ))),
+                )
+                .await
+                .map_err(rpc_err_handler)?
+                {
+                    ListOrValue::Value(t) => t.expect("value passed in; value comes out; qed"),
+                    _ => unreachable!(),
+                },
+            ),
+        }
+    }
 }
 
-#[derive(Debug, Parser)]
-struct ReplayCli {
-    #[clap(flatten)]
-    pub shared: SharedParams,
+#[cfg(not(feature = "always-wasm"))]
+pub(crate) fn build_executor<D: NativeExecutionDispatch>() -> NativeElseWasmExecutor<D> {
+    let heap_alloc_strategy = HeapAllocStrategy::Dynamic {
+        maximum_pages: Some(2048),
+    };
+    let max_runtime_instances = 8;
+    let runtime_cache_size = 2;
```

Comment:
Maybe it is worth to get the values from arguments? Insubstantial, `todo` and an issue for future is more than enuogh

Reply:
That's what the original `try-runtime-cli` does now. I thought it was insignificant for us at this point, but can just copy and paste a few extra lines to take these values as CLI arguments.

Reply:
ok